src/Sylius/Bundle/TaxonomyBundle/Doctrine/ORM/TaxonRepository.php

```diff
@@ -23,20 +22,30 @@ class TaxonRepository extends EntityRepository implements TaxonRepositoryInterfa
     /**
      * {@inheritdoc}
      */
-    public function findChildren($parentCode)
+    public function findChildren($parentCode, $isParentRoot = false)
     {
         $queryBuilder = $this->createQueryBuilder('o');
         $queryBuilder
             ->addSelect('translation')
             ->leftJoin('o.translations', 'translation')
-            ->addSelect('child')
-            ->leftJoin('o.children', 'child')
-            ->leftJoin('o.parent', 'parent')
-            ->andWhere('parent.code = :parentCode')
             ->addOrderBy('o.position')
-            ->setParameter('parentCode', $parentCode)
+            ->setParameter('code', $parentCode)
         ;
 
+        if ($isParentRoot) {
+            $queryBuilder
+                ->leftJoin('o.root', 'root')
+                ->andWhere('root.code = :code')
+            ;
+        } else {
```

Comment:
That's what I did in the first place but @lchrusciel said the method `findChildrenForRootParent` was redundant with the `findChildren` one.